crates/services/sync/src/state.rs

```diff
@@ -178,8 +190,14 @@ impl State {
     }
 
     fn apply_status(&mut self, status: Option<Status>) {
-        if let Some(s) = status {
-            self.status = s;
+        match status {
+            Some(s) => {
+                tracing::debug!("Status change from: {:?}, to: {:?}", self.status, s);
```

Comment:
@Voxelot This is actually where the event is logged but it will capture the spans from parent `instrument` functions. Would you like these to be info?

Reply:
yeah that would be helpful, as well as info-level confirmation of each block imported if the range is < 100 blocks. If we're syncing a large range of thousands of blocks it would be too noisy to log each block individually.

Reply:
I mean this does mean the block is confirmed imported because this call only happens post execute_and_commit.

Reply:
I don't think we should disable logging programmatically because it's very hard to do and also someone might need to know what when wrong even post 100 blocks. If it's too noisy that's what filtering is for.

Reply:
Ah true, if you're syncing small ranges <100 this should get logged often enough to show progress. However, even showing the latest block id that was committed from the range would be a useful improvement.

For very large ranges it would still be hard to tell how well your node is even progressing or if it stalled for some reason since this wouldn't get logged very often.

Reply:
It does show the block id though. It's just captured in the span above this not the event. So it will print out an event that inner state has changed that is within the span showing for which block.
As for adjusting log levels I'd strongly suggest for debugging the best approach is always going to be capture everything then filter through the logs after. Cutting out information for convenience means that you will potentially miss the cause of an issue. It's also possible to dynamically change the log filter without restarting the server but I think capturing and replaying / filtering is always better.

Reply:
Ok with the latest push info level trace gives:
<img width="868" alt="image" src="https://user-images.githubusercontent.com/7286454/214759348-61a5e891-80ca-432f-acd8-963297429139.png">

Reply:
vs debug level 
<img width="868" alt="image" src="https://user-images.githubusercontent.com/7286454/214759458-6fa0b5a5-d602-4415-a512-cdc1d613a90a.png">

Reply:
Nice this looks pretty good!

As far as logging in large ranges vs just enabling more debug levels, this is more of a UX issue than a debugging issue. Node operators are accustomed to seeing regular log updates when syncing a node that is far behind the network so they can see how well it's progressing. If nothing is updated in the logs for a long time they may think the sync isn't working even when it is.

Reply:
I'm not sure I understand, why would sync be making progress but the logs not update?

Reply:
let's say the range is 0..=10_000_000, if we only log when the state changes from processing to committed then it might be a long time for logs to come in. Although I guess this is mitigated by the state changing as new block heights are discovered. We can fix this later if it looks like an issue.

Reply:
It logs on each block in that range